lib/matplotlib/pyplot.py

```diff
@@ -398,7 +400,11 @@ def draw_if_interactive():
 
     # make sure the repl display hook is installed in case we become
     # interactive
-    install_repl_displayhook()
+    try:
+        install_repl_displayhook()
+    except ImportError as err:
```

Comment:
Can we catch the `NotImplemented` error here instead and make no changes to `install_repl_displayhook`?

Reply:
Sure, I changed it! In light of your next comment -- is this still relevant in general (should I amend my PR)?

Reply:
Yes, I think the technical question is if we 

 - eat the error
 - warn
 - re-raise `ImportError`

Reply:
Clearly I'm no authority on this, but warning and re-raising seem pretty logical to do because it just crashes otherwise.  

I also didn't fully understand your comment -- like why should there be no GUI at all with block=True/sleep enabled. The couple of examples I tried all seemed to work (switching to a different backend and continuing as intended). I'm very likely missing something, but is the expected behavior here something different?